app/questionnaire/variants.py

```diff
@@ -20,26 +19,16 @@ def choose_variant(
     for variant in block.get(variants_key, []):
         when_rules = variant.get("when", [])
```

Comment:
"when" for variants are mandatory.
```suggestion
        when_rules = variant["when"]
```

Reply:
Applied